src/main/java/Duke.java

```diff
@@ -1,10 +1,210 @@
+
 public class Duke {
+    // Initialise the task list
+    private static ArrayList<Task> taskList = new ArrayList<>();
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
-    }
-}
+        greetUser();
+        main();
+    }
+
+    /**
+     * Prints out the greeting message.
+     */
+    private static void greetUser() {
+        System.out.println("Hello! I'm Jukebox :)\n" + "What can I do for you today?\n");
+    }
+
+    /**
+     * Prints out the exit message.
+     */
+    private static void exitJukebox() {
+        System.out.println("Aww... OK, Hope to see you again!");
+    }
+
+    /**
+     * Main logic handler.
+     */
+    public static void main() {
+        // Variables
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+
+        // Logic handler
+        while (true) {
+            input = sc.nextLine();
+            String[] inputWordsSplit = input.split(" ", 2);
+            String commandWord = inputWordsSplit[0];
+            CommandWord command = null;
+
+            // Updating the command word, catching any invalid inputs
+            try {
+                command = CommandWord.getCommand(commandWord);
+            } catch (IllegalArgumentException e) {
+                System.out.println(e.getMessage());
+            }
+
+            // Print out goodbye message and exits the program
+            if (command == CommandWord.BYE) {
+                exitJukebox();
+                break;
+            }
+
+            // Prints out the task list
+            if (command == CommandWord.LIST) {
+                printList(taskList);
+                continue;
+            }
+
+            // Marks or unmarks the given task
+            if (command == CommandWord.MARK || command == CommandWord.UNMARK) {
+                markUnmarkTask(command, inputWordsSplit);
+                continue;
+            }
+
+            // Add a new task based on the given keyword
+            if (command == CommandWord.TODO || command == CommandWord.DEADLINE || command == CommandWord.EVENT) {
+                try {
+                    addNewTask(command, inputWordsSplit[1]);
+                    continue;
+                } catch (IndexOutOfBoundsException e) {
+                    System.out.println("Hmm... looks like you don't have a description for this task :( Please try again!\n");
+                }
+            }
+
+            // Deletes a task from the task list
+            if (command == CommandWord.DELETE) {
+                deleteTask(inputWordsSplit);
+            }
+        }
+    }
+
+
+        /**
+         * Prints out the task list iteratively.
+         * @param taskList Input task list
```

Comment:
Should there be a period at the end of your sentences? (i.e.`@param taskList Input task list.`)